server/models/visits.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var q = require('q');
+
+var config = require('../config');
+var postgres = require('../db/postgres');
+var elasticsearch = require('../db/elasticsearch');
+var log = require('../logger')('server.models.visits');
+var visit = require('./visit');
+
+var _verbose = function() {
```

Comment:
maybe a base model class with this and other copypasta'd functions like _onFulfilled could temporarily work in place of the inheritance v. composition mental debate